src/components/ha-dialog.ts

```diff
@@ -123,7 +123,13 @@ export class HaDialog extends DialogBase {
       .mdc-dialog .mdc-dialog__surface {
         position: var(--dialog-surface-position, relative);
         top: var(--dialog-surface-top);
-        margin-top: var(--dialog-surface-margin-top);
+        margin-top: calc(
+          var(--dialog-surface-margin-top) + var(--safe-area-inset-top)
+        );
+        // TODO support left, right and bottom safe area
```

Comment:
This needs to be done in content probably, this is only on mobile/narrow I guess?